clippy_lints/src/assertions_on_constants.rs

```diff
@@ -133,7 +132,7 @@ fn match_assert_with_message<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx Expr<'_>)
         if let ExprKind::Block(ref inner_block, _) = block_expr.kind;
         if let Some(begin_panic_call) = &inner_block.expr;
         // function call
-        if let Some(args) = match_function_call(cx, begin_panic_call, &paths::BEGIN_PANIC);
+        if let Some(args) = match_panic_call(cx, begin_panic_call);
```

Comment:
This change isn't necessary, since we match the exact `assert` expansion here.

Reply:
In `no_std` programs, `assert!()` doesn't expand to begin_panic.

(Also, we'll be making a changes in the standard library soon to what `assert!()` gets expanded to. It's best if clippy doesn't assume a specific panic function to be called.)